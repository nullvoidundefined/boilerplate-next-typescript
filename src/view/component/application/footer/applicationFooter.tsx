import styled from "styled-components";
import { APPLICATION_COLOR, APPLICATION_DIMENSION } from "../../../../constant";

const StyledApplicationFooter = styled.footer`
  background-color: ${APPLICATION_COLOR.FOOTER_BACKGROUND};
  height: ${APPLICATION_DIMENSION.FOOTER_HEIGHT}px;
`;

const ApplicationFooter = () => {
  return <StyledApplicationFooter>Footer</StyledApplicationFooter>;
};

export { ApplicationFooter };
