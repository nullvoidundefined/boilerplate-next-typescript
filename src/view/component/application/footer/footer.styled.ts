import styled from "styled-components";
import { APPLICATION_COLOR, APPLICATION_DIMENSION } from "../../../../constant";

const StyledFooter = styled.footer`
  background-color: ${APPLICATION_COLOR.FOOTER_BACKGROUND};
  height: ${APPLICATION_DIMENSION.FOOTER_HEIGHT}px;
`;

export { StyledFooter };
