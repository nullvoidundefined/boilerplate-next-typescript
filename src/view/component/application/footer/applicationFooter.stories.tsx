import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { ApplicationFooter } from './applicationFooter';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Component/Application/Footer',
  component: ApplicationFooter,
} as ComponentMeta<typeof ApplicationFooter>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof ApplicationFooter> = () => <ApplicationFooter />;

export const Default = Template.bind({});