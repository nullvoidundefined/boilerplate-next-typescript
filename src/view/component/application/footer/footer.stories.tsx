import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Footer } from './footer';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Component/Application/Footer',
  component: Footer,
} as ComponentMeta<typeof Footer>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Footer> = () => <Footer />;

export const Default = Template.bind({});