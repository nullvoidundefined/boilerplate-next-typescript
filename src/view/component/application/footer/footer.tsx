import { StyledFooter } from "./footer.styled";

const Footer = () => {
  return <StyledFooter>Footer</StyledFooter>;
};

export { Footer };
