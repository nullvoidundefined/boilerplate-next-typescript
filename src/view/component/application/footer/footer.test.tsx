import { Footer } from "./footer";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const footer = renderer.create(<Footer />).toJSON();
  expect(footer).toMatchSnapshot();
});
