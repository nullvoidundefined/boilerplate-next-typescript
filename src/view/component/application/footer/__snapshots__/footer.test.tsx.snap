// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="sc-ccc5a96a-0 iFPUnX"
>
  Footer
</footer>
`;
